spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/authentication/XsuaaJwtDecoder.java

```diff
@@ -47,32 +53,72 @@
 	@Override
 	public Jwt decode(String token) throws JwtException {
 		Assert.notNull(token, "token is required");
-		try {
-			JWT jwt = JWTParser.parse(token);
-			String subdomain = getSubdomain(jwt);
+		JWT jwt;
 
-			String zid = jwt.getJWTClaimsSet().getStringClaim("zid");
-			JwtDecoder decoder = cache.get(subdomain, k -> this.getDecoder(zid, subdomain));
-			return decoder.decode(token);
+		try {
+			jwt = JWTParser.parse(token);
 		} catch (ParseException ex) {
 			throw new JwtException("Error initializing JWT decoder:" + ex.getMessage());
 		}
+
+		String jku = (String) jwt.getHeader().toJSONObject().getOrDefault("jku", null);
+		String kid = (String) jwt.getHeader().toJSONObject().getOrDefault("kid", null);
+		String uaadomain = xsuaaServiceConfiguration.getUaaDomain();
+
+		try {
+			canDoOnlineVerification(jku, kid, uaadomain);
```

Comment:
Can you change the name? I understand onlineVerification as calling the /check_token endpoint. This is not meant, but rather getting the keys from /token_keys

Reply:
done, renamed to VerifyWithOnlineKey